### PR TITLE
Update to Gradle 8.1.1

### PR DESCRIPTION
**Pull request by bigdaz**

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,50 +1,17 @@
...
-dependencyResolutionManagement {
-    repositories {
-        maven {
-            name = "Gradle public repository"
-            url = uri("https://repo.gradle.org/gradle/public")
-            content {
-                includeGroup("net.rubygrapefruit")
-                includeModule("classycle", "classycle")
-                includeModule("flot", "flot")
-                includeModule("org.gradle", "gradle-tooling-api")
```
Comment (JLLeitschuh) [inline settings.gradle.kts:23]: Do we not want to resolve the tooling API from Gradle's internal artifact server?

File: settings.gradle.kts [hunk 2]
```diff
@@ -1,50 +1,17 @@
...
-            }
-        }
-        maven {
-            name = "CHAMP libs"
-            url = uri("https://releases.usethesource.io/maven/")
-            mavenContent {
-                includeGroup("io.usethesource")
-            }
-        }
-        mavenCentral()
```
Comment (JLLeitschuh) [inline settings.gradle.kts:40]: Why not leave this to be confighred for the entire build from here?

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,50 +1,17 @@
-import org.gradle.api.internal.FeaturePreviews
-
-pluginManagement {
-    plugins {
-        id("org.jetbrains.kotlin.jvm") version "1.8.10"
-        id("com.github.johnrengelman.shadow") version "8.1.0"
-    }
-}
```
Comment (JLLeitschuh) [inline settings.gradle.kts:8]: Why not keep common version declarations for plugins at the top level of the Gradle Build?

Wasn't that the model that idiomatic Gradle was pushing? Plugin versions weren't supposed to be declared in child build.gradle files IIRC.

**Review by JLLeitschuh — COMMENTED** [review]
Mostly open questions

**Comment by bigdaz** [comment]
> Mostly open questions

@JLLeitschuh This is a DRAFT PR. Please refrain from comments/review until it's ready.
I use PRs to run things on CI.

File: settings.gradle.kts (hunk 1 above)
Comment (bigdaz) [reply settings.gradle.kts:23]: I didn't debug, but it seems the `dependencyResolutionManagement` section was having no effect. All dependencies were already being resolved from `mavenCentral`. I think this is OK.
Before change: https://ge.gradle.org/s/63b23yvefr5lm/dependencies/repositories?repoToggled=W1swXSxbMV0sWzJdLFszXV0
After change: https://ge.gradle.org/s/thvqexo2mx7py/dependencies/repositories?repoToggled=W1swXV0

File: settings.gradle.kts (hunk 2 above)
Comment (bigdaz) [reply settings.gradle.kts:40]: That's a good point. I'll move the `mavenCentral` declaration back to settings.

File: settings.gradle.kts (hunk 3 above)
Comment (bigdaz) [reply settings.gradle.kts:8]: Gradle 8.1 supports `libs.plugins.X` in the plugins block. Using this mechanism seems to make the `pluginManagement` declaration unnecessary.

**Pull request closed without merging**